app/helpers/contacts_helper.rb

```diff
@@ -9,4 +9,11 @@ def start_a_conversation_link(aspect, contacts_size)
                   title: t("contacts.index.start_a_conversation")
     end
   end
+
+  def gon_load_contact(contact)
+    Gon.preloads[:contacts_collection] ||= []
+    if contact.id.nil? || !Gon.preloads[:contacts_collection].map {|contact| contact[:id] }.include?(contact.id)
```

Comment:
What's `contact.id.nil?` supposed to check for, in what situation does that happen?

For the second part, `Gon.preloads[:contacts_collection].none? {|stored_contact| stored_contact[:id] == contact.id }`

I'm also not too found of things like `collection`, `map`, `array`, `hash` in the name of things, can we just call it `contacts` or `user_contacts`?